### PR TITLE
Tsunami on landcover

### PR DESCRIPTION
**Pull request by Gustry**
Tsunami on landcover.
The semiology is not perfect as it only shows the hazard classes and not the landcover

.

### CODE REVIEW

File: safe/impact_functions/inundation/tsunami_raster_landcover/impact_function.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+__author__ = 'etiennetrimaille'
+__project_name__ = 'inasafe-dev'
+__filename__ = 'impact_function.py'
+__date__ = '11/03/16'
+__copyright__ = 'etienne@kartoza.com'
+
+
+class TsunamiRasterLandcoverFunction(ContinuousRHClassifiedVE):
+    # noinspection PyUnresolvedReferences
+    """Simple impact function for tsunami on roads."""
```
Comment (timlinux) [inline safe/impact_functions/inundation/tsunami_raster_landcover/impact_function.py]: On landcover not roads I guess...

**Comment by timlinux** [comment]
Awesome stuff. Code wise it is LGTM - lets see if @samnawi and friends can test it out and offer any feedback before we merge. Thanks @Gustry

**Comment by Gustry** [comment]
@Charlotte-Morgan I fixed your issues in this PR for GenericOnLandcover and TsunamiOnLandcover : https://github.com/inasafe/inasafe/pull/2709#issuecomment-218406426

> The prototype graph is good to see but probably does not need to be included in the detailed report. I think the table of numbers is fine.

Fixed, this PR doesn't show any plots.

> Would it be possible to make the order in teh table Low Hazard Zone on the left .. Medium and High on the right.

Fixed

> We also have a convention of omitting 0 values

Not fixed yet. I will have a look.

> We need to add notes and assumptions. If 0 values are omitted then this is usually stated in the notes.

For the notes, Tsunami On Landcover doesn't contain any notes too. See https://github.com/inasafe/inasafe/issues/2764

**Comment by samnawi** [comment]
Hi @Gustry 
I'm sorry I just try your PR now.
Overall is LGTM, the PR is running very well using QGIS 2.14. Both of vector and raster hazard.

But, I've some issue, especially in 2.8.8 :
- I cannot run your PR using this data --> [Ambon](https://www.dropbox.com/s/djezyzsafm6cmu8/Tsunami%20Ambon%20landcover.zip?dl=0)
- I can run your PR using this data --> [Maumere](https://www.dropbox.com/s/49ux7xhrpk8k1hy/Tsunami%20Maumere%20Landcover.zip?dl=0)

**Comment by felix-yew** [comment]
Hi @Gustry 

Same with @samnawi, I have tried test the tsunami hazard raster on Ambon with land cover polygon in QGIS 2.8.8, and there can not work.

**Comment by Gustry** [comment]
According to @felix-yew and @samnawi, they have a problem using this PR with QGIS 2.8.8 and some data in Ambon: 
```
Details

These additional details were reported when the problem occurred.

An exception occurred when running the impact analysis.
'NoneType' object has no attribute 'wkbType'
Diagnostics

In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_functions\base.py", line 677, in run_analysis self._impact = self._calculate_impact()
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_functions\base.py", line 1395, in _calculate_impact result_layer = self.run()
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_functions\inundation\tsunami_raster_landcover\impact_function.py", line 156, in run hazard_value_to_class, impact_fields, writer)
In file "C:/Users/kurnia/.qgis2/python/plugins\inasafe\safe\impact_functions\generic\classified_polygon_landcover\impact_function.py", line 61, in _calculate_landcover_impact if not impact_geometry.wkbType() == QGis.WKBPolygon and \
```

I tried this PR with QGIS 2.8.7, no problem for me (same dataset), I'm checking deeper.

**Comment by Gustry** [comment]
I'm sorry, I still can't replicate your issue using on QGIS 2.8.9 : 
* Ambon_Flowdepth15m_Raster_WGS84.tif
* Ambon_Landcover_BIG_WGS84.shp



We can see the `Affected column` with the percentage.

**Comment by fredychandra** [comment]
Hi @Gustry it look nice. My comment is same with my last reply on #2796

**Comment by Gustry** [comment]
As we are already in the feature freeze, is it possible to merge this PR for now ?
We can still fill tickets and fix them next week before the release.
It will allow more testing if the new Tsunami IF is in develop compare to my own branch.

File: safe/impact_template/land_cover_report_template.py [hunk 2]
```diff
@@ -103,29 +106,43 @@ def format_impact_summary(self):
...
                     filter_value=zone)
                 report['impacted_zones'][zone] = table
 
         message = m.Message(style_class='container')
         affected_text = tr('Affected Area (ha)')
 
+        show_affected = True if len(self.affected_columns) else False
+        if show_affected:
+            msg = tr(
+                '* Percentage of affected area compare to the total area for '
```
Comment (timlinux) [inline safe/impact_template/land_cover_report_template.py]: compare -> compared

**Comment by timlinux** [comment]
LGTM - please merge so we can get it broadly tested

Also for the table please add units to either columns or rows (make it consistent with other IF's) - it isn't clear what the numbers are - ha? m2? 

```In the event of tsunami how many land cover might be affected```

I wonder if we should change 'many' to 'much' here?

@lucernae and @gustry also remember that there is an expectation that we immediately start prepping the ash on landcover analysis once this IF is done so @lucernae please keep track of developments here and get the ASH on landcover ready to demo to Adele

**Pull request merged**